doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+
+        When updating a `Learner` you can only perform blocking updates on batches of data.
+        You can perform non-gradient based updates using `additional_update`.
+    
+
+Getting and setting state
+-------------------------
+
+.. tab-set::
+    
+    .. tab-item:: Getting and Setting State for a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner_group.get_state()
+            learner_group.set_state(state)
+
+            # just module weights
+            weights = learner_group.get_weights()
+            learner_group.set_weights(weights)
+
+        Set/get the state dict of all learners through learner_group via `set_state` or 
+        `get_state`. This includes all states including both neural network weights, 
+        and optimizer states on each learner.
+    
+    .. tab-item:: Getting and Setting State for a Learner
+
+        .. testcode::
+            :skipif: True
+
+            # module weights and optimizer states
+            state = learner.get_state()
+            learner.set_state(state)
+
+            # just module weights
+            weights = learner.module.get_state()
+            learner.module.set_state(weights)
+
+        You can set and get the weights of a `Learner` using `set_state` and `get_state` .
+        For setting or getting only RLModule weights (without optimizer states), use `set_weight` or `get_weight` API.
+
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+	import shutil
+	import tempfile
+
+	LEARNER_CKPT_DIR = str(tempfile.TemporaryDirectory())
+	LEARNER_GROUP_CKPT_DIR = str(tempfile.TemporaryDirectory())
+
+
+Checkpointing
+-------------
+
+.. tab-set::
+    
+    .. tab-item:: Checkpointing a LearnerGroup
+
+        .. testcode::
+            :skipif: True
+
+            learner_group.save_state(LEARNER_GROUP_CKPT_DIR)
+            learner_group.load_state(LEARNER_GROUP_CKPT_DIR)
+        
+        Checkpoint the state of all learners in the `LearnerGroup` via `save_state` and
+        `load_state`. This includes all states including neural network weights and any
+        optimizer states. Note that since the state of all of the `Learner` s is identical,
+        only the states from the first `Learner` need to be saved.
+
+    .. tab-item:: Checkpointing a `Learner`
+
+        .. testcode::
+            :skipif: True
+
+            learner.save_state(LEARNER_CKPT_DIR)
+            learner.load_state(LEARNER_CKPT_DIR)
+
+        Checkpoint the state of a `Learner` via `save_state` and `load_state`. This 
+        includes all states including neural network weights and any optimizer states.
+
+
+.. testcode::
+	:hide:
+    :skipif: True
+
+	shutil.rmtree(LEARNER_CKPT_DIR)
+	shutil.rmtree(LEARNER_GROUP_CKPT_DIR)
+
+
+Implementation
+==============
+`Learner` has many APIs for flexible implementation, however the core ones that you need to implement are:
```

Comment:
Imho we should clarify what this section deals with.
Is this about implementing a learner from scratch? Or for customization of an existing learner?

The heading should be something like "Customizing Learner" or "Implementing your own Learner" or something like that.

Reply:
why can't it be both? I feel like this might be semantics.